apps/mobile-wallet/src/features/modals/BottomModal2.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  BottomSheetBackdropProps,
+  BottomSheetModal,
+  BottomSheetModalProps,
+  BottomSheetView
+} from '@gorhom/bottom-sheet'
+import { useCallback, useEffect, useRef } from 'react'
+import { useTheme } from 'styled-components/native'
+
+import BottomModalBackdrop from '~/features/modals/BottomModalBackdrop'
+import { BottomModalBaseProps } from '~/features/modals/BottomModalBase'
+import BottomModalHandle from '~/features/modals/BottomModalHandle'
+import BottomModalHeader from '~/features/modals/BottomModalHeader'
+import { closeModal } from '~/features/modals/modalActions'
+import { useAppDispatch } from '~/hooks/redux'
+import { DEFAULT_MARGIN, VERTICAL_GAP } from '~/style/globalStyle'
+
+export type BottomModal2Props = BottomSheetModalProps & BottomModalBaseProps
+
+const BottomModal2 = ({
+  children,
+  title,
+  titleAlign,
+  navHeight,
+  modalId,
+  noPadding,
+  contentVerticalGap,
+  ...props
+}: BottomModal2Props) => {
+  const bottomSheetModalRef = useRef<BottomSheetModal>(null)
+  const dispatch = useAppDispatch()
+  const theme = useTheme()
+
+  useEffect(() => {
+    bottomSheetModalRef.current?.present()
+  }, [])
```

Comment:
In our existing modal system, the bottom modal opens when mounted. This is not the case for sheets. It only opens when explicitly calling the `present()` method.